app.go

```diff
@@ -142,7 +143,8 @@ type Headscale struct {
 	dbDebug    bool
 	privateKey *key.MachinePrivate
 
-	DERPMap *tailcfg.DERPMap
+	DERPMap            *tailcfg.DERPMap
+	EmbeddedDerpServer *EmbeddedDerpServer
```

Comment:
The term `DERP` is becoming overloaded and a bit hard to keep track of, how about:

```suggestion
	Map            *tailcfg.DERPMap
	Server *DERPServer
```

This is already in a section called `DERPConfig`.